### ISSUE TITLE
Improving existing Wordpress articles in Explore

### ISSUE THREAD

**Issue by jennpb** [wellcometrust/wellcomecollection.org#449]
To make the wordpress articles less clunky in Explore we need to investigate and change wherever possible these things.

**Option 1: hide author details on legacy content; hardcode author details for Electricity**

**Option 2: hack it**
- Russell as author for everything - can we pull author from standfirst instead? 

- To display author bio, need to include in WP and pull in through API

- To display author twitter handle, need to include in WP and pull in through API

**Comment by jennpb** [comment]
@RussellDornan @dannybirchall What do you think about option 1? this would allow us to get to release sooner. We'd then pick up author names and other extended info once we get to looking at content creation tools.

**Comment by dannybirchall** [comment]
@jennpb @RussellDornan well, everything's possible -- but what's the timeframe for content creation tools? I think the reconstructed byline and standfirst are some of the strongest parts of the new design -- I'd probably prefer a hack to maker it look nice, even if it's clunky under the cover.

**Comment by dannybirchall** [comment]
(though would prefer not to have Russell as author for everything -- sorry @RussellDornan!)

**Comment by RussellDornan** [comment]
@jennpb @dannybirchall I don't want to be the author for everything ever again!

**Comment by jennpb** [comment]
@dannybirchall @RussellDornan I conferred with @jamesgorrie on this. For the first pass, I think the best option is to hide the author details UNLESS actually written by RD himself. The author isn't always in teh standfirst, which complicates the hack. And we don't have avatars for all guests. To do it manually, we'd need a list of all articles' authors, but it'd still be a hack that we'd have to roll back later.

We will include the author and author bio for Electricity content, so think about getting Ruth's photo and bio :)

**Comment by dannybirchall** [comment]
What if we faked up the authors on wordpress?

**Comment by jennpb** [comment]
Wouldn't you need unique email addresses for each author?

**Comment by dannybirchall** [comment]
Oh, ugh, yes you do.

**Comment by jamesgorrie** [comment]
> faked up the authors on wordpress

Do you mean just create fake Authors? That could work . but as @jennpb said, emails might be a hitch.

The other "clunky under the cover" option is to have a map of slug => authors stored (and hard-coded) in the codebase... we then show the author if we have one, or don't if we don't (or even better show @RussellDornan if we don't).

I'll knock this up quick.

**Comment by jennpb** [comment]
And would also need to deal wtih redundancy

![image](https://cloud.githubusercontent.com/assets/1446139/22692618/b48b5ea4-ed37-11e6-8016-47eb58e133f7.png)

**Comment by dannybirchall** [comment]
So we can go back and edit standfirsts -- I think having access to author functionality in new posts trumps duplication in old posts.

**Comment by jennpb** [comment]
Could we hide on old posts and show on new posts, with each new author added to WP?

**Comment by jamesgorrie** [comment]
We can have a fake system (we hard code the authors), or the Wordpress system.
If we want it for all new posts, we just make sure we hard-code those in. Simple.

Or we add the authors to WP, and leave it as is.

I'm going to go for the former as it allows us to  publish correctly with Electricity, we can retrospectively add more when we want, and helps with the migration once we get there.

**Comment by dannybirchall** [comment]
Ok (I think) -- you might have to explain how the hardcoding would work.

(Did we dispense with the idea of doing the fake using tags/categories in wordpress?)

**Comment by jamesgorrie** [comment]
@dannybirchall @jennpb 

To hard code it - we would need to add authors in [this syntax](https://github.com/wellcometrust/wellcomecollection.org/blob/author-lookup/server/data/people.js).

We would then need to assign each article [here](https://github.com/wellcometrust/wellcomecollection.org/blob/author-lookup/server/services/author-lookup.js).

This seems to me like it would be easier than the solution around creating new authors in wordpress, or tagging articles up with a special tag... 

What do you think?

I have tried to find a way to pull this info out automatically, but can't find a common pattern etc in the way we say someone is an author of an article.

**Comment by jennpb** [comment]
The work to extract authors would need to be tackled by someone outside of our team. @dannybirchall Do you know of anyone who could do this?

@jamesgorrie and I both feel this is a better option than setting up new WP accounts under fake gmail addresses - the WP account would need to be activated for each author, meaning someone would have to log in to each and every fake gmail account ...that's a lot of work. Adding the auths and articles directly to the JSON is the most efficient way of doing it.

**Comment by dannybirchall** [comment]
Hi @jamesgorrie @jennpb 

Hmmm, I'm not sure that hardcoding data structures is 'easier' from an inputting-the-actual-data point of view, but I'm willing to give whatever works a go, if @RussellDornan is too.

Where would we put this then?

**Comment by dannybirchall** [comment]
Also, the bit I'm struggling slightly with is -- is it necessary to retrospectively fix authorship for all previous authors in order to have nice author display for new posts?

**Comment by dannybirchall** [comment]
Also, @jamesgorrie the data structure doesn't have a twitter handle -- should it?

**Comment by jamesgorrie** [comment]
@dannybirchall so far as I can tell we have three options:
1. - Create WP authors with fake accounts
    - Validate those accounts
    - Apply those users to the posts
2. - Add authors to JS model
    - Map authors to slugs in WP
3. - Create tags on articles that follow a pattern e.g. author: FirstName LastName twitterhandle

With option `2` - it's basically if you can give us some structured data of authors =>slug, we can always write a script to turn it into the JS model.

**Comment by dannybirchall** [comment]
@jamesgorrie think I need to come talk to you about this.

**Comment by jennpb** [comment]
We decided to do the following
- Show author/twitter handle only when true author = Russell (https://next.wellcomecollection.org/articles/oops-i-wrote-a-britney-blog-post/)
- Hide author/twitter handle when different author found in standfirst.
- Hardcode Ruth as Electricity author 

All else contingent on a new CMS!

**Comment by dannybirchall** [comment]
Oh.

Is it possible to extend what Russell has to me, Lalita and the new editor (all of whom'll have WP accounts)? If we'll be using the blog as CMS for a while, it seems unfair to only have Russell as a named author.

Will we be hardcoding other digital stories editors?

What are the rules for including authors in standfirsts so they'll be recognised?

Does this make the slug/author spreadsheet we're doing redundant?

**Comment by jamesgorrie** [comment]
> it seems unfair to only have Russell as a named author.

The point of adding the slug => author mapping so this is not the case.
We've already done this for Ruth Garde.

> Will we be hardcoding other digital stories editors?

Yes.

> What are the rules for including authors in standfirsts so they'll be recognised?

I imagine we'd like to give the Authors and expose those on the frontend as per the design.

If we could get the info for Sarah Bentley who wrote [this](https://blog.wellcomecollection.org/2017/02/15/the-ladies-of-llangollen/) (twitterhandle, picture and bio) we could show you what that will be like.

> Does this make the slug/author spreadsheet we're doing redundant?

No - as this is what we need to add authors to the site.

**Comment by dannybirchall** [comment]
Ok, I'm thoroughly confused now.

**Comment by jamesgorrie** [comment]
For clarity to all:

We are hardcoding slugs => authors.
If we don't have a record of a slug => author we don't show the author.
The spreadsheet will help us populate this.
